src/modules/stackoverflow/stackoverflow.reducer.js

```diff
@@ -0,0 +1,39 @@
+import types from "./stackoverflow.types";
+
+const initState = {
+  downloadedData: [],
+  messages: [],
+  topic: "",
+};
+const reducer = (state = initState, action) => {
+  switch (action.type) {
+    case types.SET_TOPIC:
+      //console.log(action.payload.topic);
+      return { ...state, topic: action.payload.topic };
+    case types.SET_DATA:
+      //console.log(action.payload.downloadedData);
+      return { ...state, downloadedData: action.payload.downloadedData.items };
+    case types.SORT_DATA:
+      console.log("kot w reduc");
+      function sortNum(a, b) {
+        if (action.payload.sortingKey === "creation_date") {
+          return a.creation_date - b.creation_date;
+        } else {
+          return a.owner.reputation - b.owner.reputation;
+        }
+      }
+      return {
+        ...state,
+        downloadedData: state.downloadedData.sort(sortNum),
+      };
```

Comment:
Nie jest to złe rozwiązanie jednak musimy pamiętać, aby zrobić najpierw kopie sortowanych elementów bo `.sort` modyfikuje tablicę, po której sortowanie więc: `[...state.downloadedData].sort(...)`